src/app/utils/field-validation/validators/tableValidator.ts

```diff
@@ -0,0 +1,146 @@
+import { createAnswerFieldFromColumn } from '../answerField.factory'
+import { validateField } from '..'
+
+const ALLOWED_COLUMN_TYPES = [BasicField.ShortText, BasicField.Dropdown]
+
+type TableValidator = ResponseValidator<ProcessedTableResponse>
+type TableValidatorConstructor = (
+  tableField: ITableFieldSchema,
+) => TableValidator
+
+/**
+ * Returns a validation function to check if the
+ * response has less than the minimum number of rows specified.
+ */
+const makeMinimumRowsValidator: TableValidatorConstructor = (tableField) => (
+  response,
+) => {
+  const { answerArray } = response
+  const { minimumRows } = tableField
+
+  return answerArray.length >= minimumRows
+    ? right(response)
+    : left(`TableValidator:\tanswer has less than the minimum number of rows`)
+}
+
+/**
+ * Returns a validation function to check if addMoreRows is not set
+ * and if so, whether the response has more than minimum number of rows.
+ */
+const makeAddMoreRowsValidator: TableValidatorConstructor = (tableField) => (
+  response,
+) => {
+  const { answerArray } = response
+  const { minimumRows, addMoreRows } = tableField
+
+  if (addMoreRows) return right(response)
+  return answerArray.length === minimumRows
+    ? right(response)
+    : left(
+        `TableValidator:\tanswer has extra rows even though addMoreRows is false`,
+      )
+}
+
+/**
+ * Returns a validation function to check if the
+ * response has more than the maximum number of rows specified.
+ */
+const makeMaximumRowsValidator: TableValidatorConstructor = (tableField) => (
+  response,
+) => {
+  const { answerArray } = response
+  const { maximumRows } = tableField
+
+  if (!maximumRows) return right(response)
+
+  return answerArray.length <= maximumRows
+    ? right(response)
+    : left(`TableValidator:\tanswer has more than the maximum number of rows`)
+}
+
+/**
+ * Returns a validation function to check if the
+ * response has the correct number of answers for each row.
+ */
+const makeRowLengthValidator: TableValidatorConstructor = (tableField) => (
+  response,
+) => {
+  const { answerArray } = response
+  const { columns } = tableField
+
+  return answerArray.every((row) => row.length === columns.length)
+    ? right(response)
+    : left(`TableValidator:\tanswer has rows with incorrect number of answers`)
+}
+
+/**
+ * Returns a validation function that checks that the columns only have allowed types
+ */
+
+const makeColumnTypeValidator: TableValidatorConstructor = (tableField) => (
+  response,
+) => {
+  const { columns } = tableField
+  return columns.every((column) =>
+    ALLOWED_COLUMN_TYPES.includes(column.columnType),
+  )
+    ? right(response)
+    : left(`TableValidator:\tanswer has columns with non-allowed types`)
+}
+
+/**
+ * Returns a validation function that applies
+ * the correct validator for each column based on its type.
+ */
+
+const makeColumnValidator: TableValidatorConstructor = (tableField) => (
```

Comment:
i think the function name is a little misleading, seems like it is validating each element in _row major order_.

Reply:
renamed